Authentication test occasionally fails

This test at times fails, especially in CI but at times also when running locally:

https://github.com/dana-team/platform-backend/blob/026b20d09299c3bd9581fe9970cf5c1fcf7ae29e/src/auth/openshift_test.go#L53-L62

Full trace from CI:

```
2024-06-15T10:48:29.0027914Z go test -v $(go list ./...) -coverprofile cover.out
2024-06-15T10:48:38.3832246Z github.com/dana-team/platform-backend/cmd coverage: 0.0% of statements
2024-06-15T10:48:38.3840552Z github.com/dana-team/platform-backend/src/controllers coverage: 0.0% of statements
2024-06-15T10:48:38.4024264Z === RUN TestObtainOpenshiftToken
2024-06-15T10:48:38.4025476Z === RUN TestObtainOpenshiftToken/ShouldFailWithUnauthorized
2024-06-15T10:48:38.4098312Z [GIN-debug] [WARNING] Running in "debug" mode. Switch to "release" mode in production.
2024-06-15T10:48:38.4100226Z - using env: export GIN_MODE=release
2024-06-15T10:48:38.4100988Z - using code: gin.SetMode(gin.ReleaseMode)
2024-06-15T10:48:38.4101433Z
2024-06-15T10:48:38.4305318Z {"level":"error","ts":1718448518.3930583,"caller":"auth/openshift.go:53","msg":"failed to check AuthCode response","error":"invalid credentials","stacktrace":"github.com/dana-team/platform-backend/src/auth.ObtainOpenshiftToken\n\t/home/runner/work/platform-backend/platform-backend/src/auth/openshift.go:53\ngithub.com/dana-team/platform-backend/src/auth.TestObtainOpenshiftToken.func2\n\t/home/runner/work/platform-backend/platform-backend/src/auth/openshift_test.go:102\ntesting.tRunner\n\t/opt/hostedtoolcache/go/1.22.2/x64/src/testing/testing.go:1689"}
2024-06-15T10:48:38.4310607Z === RUN TestObtainOpenshiftToken/ShouldFailWithFailedRequest
2024-06-15T10:48:38.4312131Z [GIN-debug] [WARNING] Running in "debug" mode. Switch to "release" mode in production.
2024-06-15T10:48:38.4313213Z - using env: export GIN_MODE=release
2024-06-15T10:48:38.4313927Z - using code: gin.SetMode(gin.ReleaseMode)
2024-06-15T10:48:38.4314391Z
2024-06-15T10:48:38.4319979Z {"level":"error","ts":1718448518.393608,"caller":"auth/openshift.go:53","msg":"failed to check AuthCode response","error":"unexpected status code: 500, body: Internal Server Error","stacktrace":"github.com/dana-team/platform-backend/src/auth.ObtainOpenshiftToken\n\t/home/runner/work/platform-backend/platform-backend/src/auth/openshift.go:53\ngithub.com/dana-team/platform-backend/src/auth.TestObtainOpenshiftToken.func2\n\t/home/runner/work/platform-backend/platform-backend/src/auth/openshift_test.go:102\ntesting.tRunner\n\t/opt/hostedtoolcache/go/1.22.2/x64/src/testing/testing.go:1689"}
2024-06-15T10:48:38.4325343Z === RUN TestObtainOpenshiftToken/ShouldSuccessObtainingToken
2024-06-15T10:48:38.4326630Z [GIN-debug] [WARNING] Running in "debug" mode. Switch to "release" mode in production.
2024-06-15T10:48:38.4327703Z - using env: export GIN_MODE=release
2024-06-15T10:48:38.4328423Z - using code: gin.SetMode(gin.ReleaseMode)
2024-06-15T10:48:38.4328870Z
2024-06-15T10:48:38.4338107Z {"level":"error","ts":1718448518.3939896,"caller":"auth/openshift.go:53","msg":"failed to check AuthCode response","error":"unexpected status code: 500, body: Internal Server Error","stacktrace":"github.com/dana-team/platform-backend/src/auth.ObtainOpenshiftToken\n\t/home/runner/work/platform-backend/platform-backend/src/auth/openshift.go:53\ngithub.com/dana-team/platform-backend/src/auth.TestObtainOpenshiftToken.func2\n\t/home/runner/work/platform-backend/platform-backend/src/auth/openshift_test.go:102\ntesting.tRunner\n\t/opt/hostedtoolcache/go/1.22.2/x64/src/testing/testing.go:1689"}
2024-06-15T10:48:38.4344447Z openshift_test.go:104: ObtainOpenshiftToken() unexpected error: unexpected status code: 500, body: Internal Server Error
2024-06-15T10:48:38.4345872Z --- FAIL: TestObtainOpenshiftToken (0.01s)
2024-06-15T10:48:38.4347312Z --- PASS: TestObtainOpenshiftToken/ShouldFailWithUnauthorized (0.00s)
2024-06-15T10:48:38.4348731Z --- PASS: TestObtainOpenshiftToken/ShouldFailWithFailedRequest (0.00s)
2024-06-15T10:48:38.4350098Z --- FAIL: TestObtainOpenshiftToken/ShouldSuccessObtainingToken (0.00s)
2024-06-15T10:48:38.4351016Z === RUN TestObtainOpenshiftUsername
2024-06-15T10:48:38.4351860Z === RUN TestObtainOpenshiftUsername/ShouldSuccessObtainingUsername
2024-06-15T10:48:38.4352915Z === RUN TestObtainOpenshiftUsername/ShouldFailWithUnauthorized
2024-06-15T10:48:38.4359101Z {"level":"error","ts":1718448518.3964093,"caller":"auth/openshift.go:70","msg":"failed to fetch Openshift user info","error":"failed to fetch userinfo, status code: 401","stacktrace":"github.com/dana-team/platform-backend/src/auth.ObtainOpenshiftUsername\n\t/home/runner/work/platform-backend/platform-backend/src/auth/openshift.go:70\ngithub.com/dana-team/platform-backend/src/auth.TestObtainOpenshiftUsername.func2\n\t/home/runner/work/platform-backend/platform-backend/src/auth/openshift_test.go:172\ntesting.tRunner\n\t/opt/hostedtoolcache/go/1.22.2/x64/src/testing/testing.go:1689"}
2024-06-15T10:48:38.4364283Z --- PASS: TestObtainOpenshiftUsername (0.00s)
2024-06-15T10:48:38.4365516Z --- PASS: TestObtainOpenshiftUsername/ShouldSuccessObtainingUsername (0.00s)
2024-06-15T10:48:38.4366963Z --- PASS: TestObtainOpenshiftUsername/ShouldFailWithUnauthorized (0.00s)
2024-06-15T10:48:38.4367785Z FAIL
2024-06-15T10:48:38.4368164Z coverage: 61.5% of statements
2024-06-15T10:48:38.4368978Z FAIL github.com/dana-team/platform-backend/src/auth 0.018s
2024-06-15T10:48:38.4369752Z === RUN TestTokenAuthMiddleware
2024-06-15T10:48:38.4370820Z [GIN-debug] [WARNING] Running in "debug" mode. Switch to "release" mode in production.
2024-06-15T10:48:38.4372089Z - using env: export GIN_MODE=release
2024-06-15T10:48:38.4372860Z - using code: gin.SetMode(gin.ReleaseMode)
2024-06-15T10:48:38.4373312Z
2024-06-15T10:48:38.4374501Z [GIN-debug] GET /ping --> github.com/dana-team/platform-backend/src/middleware.TestTokenAuthMiddleware.func2 (3 handlers)
2024-06-15T10:48:38.4375989Z === RUN TestTokenAuthMiddleware/ShouldSuccessWithValidToken
2024-06-15T10:48:38.4377035Z === RUN TestTokenAuthMiddleware/ShouldFailWithoutAuthorizationHeader
2024-06-15T10:48:38.4378881Z 2024-06-15T10:48:38.406Z ERROR middleware/auth_middleware.go:29 Failed to obtain OpenShift token {"error": "authorization token not provided"}
2024-06-15T10:48:38.4381061Z github.com/dana-team/platform-backend/src/middleware.TestTokenAuthMiddleware.TokenAuthMiddleware.func4
2024-06-15T10:48:38.4382839Z /home/runner/work/platform-backend/platform-backend/src/middleware/auth_middleware.go:29
2024-06-15T10:48:38.4424696Z github.com/gin-gonic/gin.(*Context).Next
2024-06-15T10:48:38.4425835Z /home/runner/go/pkg/mod/github.com/gin-gonic/gin@v1.10.0/context.go:185
2024-06-15T10:48:38.4427252Z github.com/dana-team/platform-backend/src/middleware.TestTokenAuthMiddleware.func1
2024-06-15T10:48:38.4428962Z /home/runner/work/platform-backend/platform-backend/src/middleware/auth_middleware_test.go:37
2024-06-15T10:48:38.4430151Z github.com/gin-gonic/gin.(*Context).Next
2024-06-15T10:48:38.4431243Z /home/runner/go/pkg/mod/github.com/gin-gonic/gin@v1.10.0/context.go:185
2024-06-15T10:48:38.4432346Z github.com/gin-gonic/gin.(*Engine).handleHTTPRequest
2024-06-15T10:48:38.4433474Z /home/runner/go/pkg/mod/github.com/gin-gonic/gin@v1.10.0/gin.go:633
2024-06-15T10:48:38.4434465Z github.com/gin-gonic/gin.(*Engine).ServeHTTP
2024-06-15T10:48:38.4435512Z /home/runner/go/pkg/mod/github.com/gin-gonic/gin@v1.10.0/gin.go:589
2024-06-15T10:48:38.4448907Z github.com/dana-team/platform-backend/src/middleware.TestTokenAuthMiddleware.func3
2024-06-15T10:48:38.4450628Z /home/runner/work/platform-backend/platform-backend/src/middleware/auth_middleware_test.go:98
2024-06-15T10:48:38.4452337Z testing.tRunner
2024-06-15T10:48:38.4453034Z /opt/hostedtoolcache/go/1.22.2/x64/src/testing/testing.go:1689
2024-06-15T10:48:38.4454510Z === RUN TestTokenAuthMiddleware/ShouldFailWithInvalidAuthorizationHeader
2024-06-15T10:48:38.4456559Z 2024-06-15T10:48:38.406Z ERROR middleware/auth_middleware.go:29 Failed to obtain OpenShift token {"error": "invalid authentication token"}
2024-06-15T10:48:38.4458691Z github.com/dana-team/platform-backend/src/middleware.TestTokenAuthMiddleware.TokenAuthMiddleware.func4
2024-06-15T10:48:38.4460490Z /home/runner/work/platform-backend/platform-backend/src/middleware/auth_middleware.go:29
2024-06-15T10:48:38.4461614Z github.com/gin-gonic/gin.(*Context).Next
2024-06-15T10:48:38.4462664Z /home/runner/go/pkg/mod/github.com/gin-gonic/gin@v1.10.0/context.go:185
2024-06-15T10:48:38.4464061Z github.com/dana-team/platform-backend/src/middleware.TestTokenAuthMiddleware.func1
2024-06-15T10:48:38.4465724Z /home/runner/work/platform-backend/platform-backend/src/middleware/auth_middleware_test.go:37
2024-06-15T10:48:38.4466937Z github.com/gin-gonic/gin.(*Context).Next
2024-06-15T10:48:38.4467980Z /home/runner/go/pkg/mod/github.com/gin-gonic/gin@v1.10.0/context.go:185
2024-06-15T10:48:38.4469063Z github.com/gin-gonic/gin.(*Engine).handleHTTPRequest
2024-06-15T10:48:38.4470161Z /home/runner/go/pkg/mod/github.com/gin-gonic/gin@v1.10.0/gin.go:633
2024-06-15T10:48:38.4471125Z github.com/gin-gonic/gin.(*Engine).ServeHTTP
2024-06-15T10:48:38.4472167Z /home/runner/go/pkg/mod/github.com/gin-gonic/gin@v1.10.0/gin.go:589
2024-06-15T10:48:38.4473497Z github.com/dana-team/platform-backend/src/middleware.TestTokenAuthMiddleware.func3
2024-06-15T10:48:38.4475087Z /home/runner/work/platform-backend/platform-backend/src/middleware/auth_middleware_test.go:98
2024-06-15T10:48:38.4476121Z testing.tRunner
2024-06-15T10:48:38.4476795Z /opt/hostedtoolcache/go/1.22.2/x64/src/testing/testing.go:1689
2024-06-15T10:48:38.4477724Z --- PASS: TestTokenAuthMiddleware (0.00s)
2024-06-15T10:48:38.4478814Z --- PASS: TestTokenAuthMiddleware/ShouldSuccessWithValidToken (0.00s)
2024-06-15T10:48:38.4480282Z --- PASS: TestTokenAuthMiddleware/ShouldFailWithoutAuthorizationHeader (0.00s)
2024-06-15T10:48:38.4481880Z --- PASS: TestTokenAuthMiddleware/ShouldFailWithInvalidAuthorizationHeader (0.00s)
2024-06-15T10:48:38.4574619Z === RUN TestLoggerMiddleware
2024-06-15T10:48:38.4575933Z [GIN-debug] [WARNING] Running in "debug" mode. Switch to "release" mode in production.
2024-06-15T10:48:38.4577051Z - using env: export GIN_MODE=release
2024-06-15T10:48:38.4577799Z - using code: gin.SetMode(gin.ReleaseMode)
2024-06-15T10:48:38.4578275Z
2024-06-15T10:48:38.4579396Z [GIN-debug] GET /ping --> github.com/dana-team/platform-backend/src/middleware.TestLoggerMiddleware.func1 (2 handlers)
2024-06-15T10:48:38.4580755Z === RUN TestLoggerMiddleware/ShouldSuccess
2024-06-15T10:48:38.4581936Z {"level":"info","ts":1718448518.4069688,"caller":"middleware/logger_middleware_test.go:25","msg":"Ping endpoint hit"}
2024-06-15T10:48:38.4583185Z --- PASS: TestLoggerMiddleware (0.00s)
2024-06-15T10:48:38.4584093Z --- PASS: TestLoggerMiddleware/ShouldSuccess (0.00s)
2024-06-15T10:48:38.4584736Z PASS
2024-06-15T10:48:38.4585102Z coverage: 76.0% of statements
2024-06-15T10:48:38.4586220Z ok github.com/dana-team/platform-backend/src/middleware 0.029s coverage: 76.0% of statements
2024-06-15T10:48:39.9422356Z ? github.com/dana-team/platform-backend/src/types [no test files]
2024-06-15T10:48:39.9438081Z github.com/dana-team/platform-backend/src/utils coverage: 0.0% of statements
2024-06-15T10:48:39.9705213Z [GIN-debug] [WARNING] Creating an Engine instance with the Logger and Recovery middleware already attached.
2024-06-15T10:48:39.9706231Z
2024-06-15T10:48:39.9707004Z [GIN-debug] [WARNING] Running in "debug" mode. Switch to "release" mode in production.
2024-06-15T10:48:39.9708118Z - using env: export GIN_MODE=release
2024-06-15T10:48:39.9709334Z - using code: gin.SetMode(gin.ReleaseMode)
2024-06-15T10:48:39.9709823Z
2024-06-15T10:48:39.9711720Z [GIN-debug] GET /v1/namespaces/ --> github.com/dana-team/platform-backend/src/routes/v1_test.SetupRouter.ListNamespaces.namespaceHandler.func11 (4 handlers)
2024-06-15T10:48:39.9714780Z [GIN-debug] GET /v1/namespaces/
--> github.com/dana-team/platform-backend/src/routes/v1_test.SetupRouter.GetNamespace.namespaceHandler.func12 (4 handlers)
2024-06-15T10:48:39.9717703Z [GIN-debug] POST /v1/namespaces/ --> github.com/dana-team/platform-backend/src/routes/v1_test.SetupRouter.CreateNamespace.namespaceHandler.func13 (4 handlers)
2024-06-15T10:48:39.9720661Z [GIN-debug] DELETE /v1/namespaces/
--> github.com/dana-team/platform-backend/src/routes/v1_test.SetupRouter.DeleteNamespace.namespaceHandler.func14 (4 handlers)
2024-06-15T10:48:39.9723736Z [GIN-debug] POST /v1/namespaces/
/secrets/ --> github.com/dana-team/platform-backend/src/routes/v1_test.SetupRouter.CreateSecret.secretHandler.func15 (4 handlers)
2024-06-15T10:48:39.9726739Z [GIN-debug] GET /v1/namespaces/
/secrets/ --> github.com/dana-team/platform-backend/src/routes/v1_test.SetupRouter.GetSecrets.secretHandler.func16 (4 handlers)
2024-06-15T10:48:39.9729783Z [GIN-debug] GET /v1/namespaces/
/secrets/
--> github.com/dana-team/platform-backend/src/routes/v1_test.SetupRouter.GetSecret.secretHandler.func17 (4 handlers)
2024-06-15T10:48:39.9733216Z [GIN-debug] PATCH /v1/namespaces/
/secrets/
--> github.com/dana-team/platform-backend/src/routes/v1_test.SetupRouter.PatchSecret.secretHandler.func18 (4 handlers)
2024-06-15T10:48:39.9739050Z [GIN-debug] DELETE /v1/namespaces/
/secrets/
--> github.com/dana-team/platform-backend/src/routes/v1_test.SetupRouter.DeleteSecret.secretHandler.func19 (4 handlers)
2024-06-15T10:48:39.9740320Z === RUN Test_Login
2024-06-15T10:48:39.9740686Z === RUN Test_Login/ShouldFailWithInternalServerError
2024-06-15T10:48:39.9741431Z [GIN-debug] [WARNING] Running in "debug" mode. Switch to "release" mode in production.
2024-06-15T10:48:39.9742479Z - using env: export GIN_MODE=release
2024-06-15T10:48:39.9743177Z - using code: gin.SetMode(gin.ReleaseMode)
2024-06-15T10:48:39.9743442Z
2024-06-15T10:48:39.9744038Z [GIN-debug] POST /login --> github.com/dana-team/platform-backend/src/routes/v1.Test_Login.func1.Login.2 (2 handlers)
2024-06-15T10:48:39.9745591Z 2024-06-15T10:48:39.956Z ERROR v1/login.go:46 Failed to obtain OpenShift token {"user": "user", "error": "some internal error"}
2024-06-15T10:48:39.9747255Z github.com/dana-team/platform-backend/src/routes/v1.Test_Login.func1.Login.2
2024-06-15T10:48:39.9748611Z /home/runner/work/platform-backend/platform-backend/src/routes/v1/login.go:46
2024-06-15T10:48:39.9749607Z github.com/gin-gonic/gin.(*Context).Next
2024-06-15T10:48:39.9750652Z /home/runner/go/pkg/mod/github.com/gin-gonic/gin@v1.10.0/context.go:185
2024-06-15T10:48:39.9752036Z github.com/dana-team/platform-backend/src/routes/v1.Test_Login.func1.LoggerMiddleware.1
2024-06-15T10:48:39.9753630Z /home/runner/work/platform-backend/platform-backend/src/middleware/logger_middleware.go:14
2024-06-15T10:48:39.9754724Z github.com/gin-gonic/gin.(*Context).Next
2024-06-15T10:48:39.9755570Z /home/runner/go/pkg/mod/github.com/gin-gonic/gin@v1.10.0/context.go:185
2024-06-15T10:48:39.9756172Z github.com/gin-gonic/gin.(*Engine).handleHTTPRequest
2024-06-15T10:48:39.9756883Z /home/runner/go/pkg/mod/github.com/gin-gonic/gin@v1.10.0/gin.go:633
2024-06-15T10:48:39.9757424Z github.com/gin-gonic/gin.(*Engine).ServeHTTP
2024-06-15T10:48:39.9757991Z /home/runner/go/pkg/mod/github.com/gin-gonic/gin@v1.10.0/gin.go:589
2024-06-15T10:48:39.9758638Z github.com/dana-team/platform-backend/src/routes/v1.Test_Login.func1
2024-06-15T10:48:39.9759420Z /home/runner/work/platform-backend/platform-backend/src/routes/v1/login_test.go:109
2024-06-15T10:48:39.9760250Z testing.tRunner
2024-06-15T10:48:39.9760629Z /opt/hostedtoolcache/go/1.22.2/x64/src/testing/testing.go:1689
2024-06-15T10:48:39.9761113Z === RUN Test_Login/ShouldSuccessObtainingToken
2024-06-15T10:48:39.9761898Z [GIN-debug] [WARNING] Running in "debug" mode. Switch to "release" mode in production.
2024-06-15T10:48:39.9762497Z - using env: export GIN_MODE=release
2024-06-15T10:48:39.9762901Z - using code: gin.SetMode(gin.ReleaseMode)
2024-06-15T10:48:39.9763159Z
2024-06-15T10:48:39.9763743Z [GIN-debug] POST /login --> github.com/dana-team/platform-backend/src/routes/v1.Test_Login.func1.Login.2 (2 handlers)
2024-06-15T10:48:39.9764477Z === RUN Test_Login/ShouldFailWithInvalidPayload
2024-06-15T10:48:39.9765106Z [GIN-debug] [WARNING] Running in "debug" mode. Switch to "release" mode in production.
2024-06-15T10:48:39.9765690Z - using env: export GIN_MODE=release
2024-06-15T10:48:39.9766094Z - using code: gin.SetMode(gin.ReleaseMode)
2024-06-15T10:48:39.9766350Z
2024-06-15T10:48:39.9766934Z [GIN-debug] POST /login --> github.com/dana-team/platform-backend/src/routes/v1.Test_Login.func1.Login.2 (2 handlers)
2024-06-15T10:48:39.9768207Z 2024-06-15T10:48:39.956Z ERROR v1/login.go:34 Invalid request payload {"error": "json: cannot unmarshal string into Go value of type types.Credentials"}
2024-06-15T10:48:39.9769234Z github.com/dana-team/platform-backend/src/routes/v1.Test_Login.func1.Login.2
2024-06-15T10:48:39.9770008Z /home/runner/work/platform-backend/platform-backend/src/routes/v1/login.go:34
2024-06-15T10:48:39.9770566Z github.com/gin-gonic/gin.(*Context).Next
2024-06-15T10:48:39.9771143Z /home/runner/go/pkg/mod/github.com/gin-gonic/gin@v1.10.0/context.go:185
2024-06-15T10:48:39.9772143Z github.com/dana-team/platform-backend/src/routes/v1.Test_Login.func1.LoggerMiddleware.1
2024-06-15T10:48:39.9773063Z /home/runner/work/platform-backend/platform-backend/src/middleware/logger_middleware.go:14
2024-06-15T10:48:39.9773690Z github.com/gin-gonic/gin.(*Context).Next
2024-06-15T10:48:39.9774259Z /home/runner/go/pkg/mod/github.com/gin-gonic/gin@v1.10.0/context.go:185
2024-06-15T10:48:39.9774848Z github.com/gin-gonic/gin.(*Engine).handleHTTPRequest
2024-06-15T10:48:39.9775445Z /home/runner/go/pkg/mod/github.com/gin-gonic/gin@v1.10.0/gin.go:633
2024-06-15T10:48:39.9775973Z github.com/gin-gonic/gin.(*Engine).ServeHTTP
2024-06-15T10:48:39.9776613Z /home/runner/go/pkg/mod/github.com/gin-gonic/gin@v1.10.0/gin.go:589
2024-06-15T10:48:39.9777257Z github.com/dana-team/platform-backend/src/routes/v1.Test_Login.func1
2024-06-15T10:48:39.9778023Z /home/runner/work/platform-backend/platform-backend/src/routes/v1/login_test.go:109
2024-06-15T10:48:39.9778550Z testing.tRunner
2024-06-15T10:48:39.9778918Z /opt/hostedtoolcache/go/1.22.2/x64/src/testing/testing.go:1689
2024-06-15T10:48:39.9779423Z === RUN Test_Login/ShouldFailWithInvalidCredentials
2024-06-15T10:48:39.9780082Z [GIN-debug] [WARNING] Running in "debug" mode. Switch to "release" mode in production.
2024-06-15T10:48:39.9780667Z - using env: export GIN_MODE=release
2024-06-15T10:48:39.9781074Z - using code: gin.SetMode(gin.ReleaseMode)
2024-06-15T10:48:39.9781327Z
2024-06-15T10:48:39.9781912Z [GIN-debug] POST /login --> github.com/dana-team/platform-backend/src/routes/v1.Test_Login.func1.Login.2 (2 handlers)
2024-06-15T10:48:39.9783114Z 2024-06-15T10:48:39.957Z WARN v1/login.go:43 Invalid credentials provided {"user": "invalid_user", "error": "invalid credentials"}
2024-06-15T10:48:39.9784043Z github.com/dana-team/platform-backend/src/routes/v1.Test_Login.func1.Login.2
2024-06-15T10:48:39.9784835Z /home/runner/work/platform-backend/platform-backend/src/routes/v1/login.go:43
2024-06-15T10:48:39.9785401Z github.com/gin-gonic/gin.(*Context).Next
2024-06-15T10:48:39.9785978Z /home/runner/go/pkg/mod/github.com/gin-gonic/gin@v1.10.0/context.go:185
2024-06-15T10:48:39.9786763Z github.com/dana-team/platform-backend/src/routes/v1.Test_Login.func1.LoggerMiddleware.1
2024-06-15T10:48:39.9787866Z /home/runner/work/platform-backend/platform-backend/src/middleware/logger_middleware.go:14
2024-06-15T10:48:39.9788515Z github.com/gin-gonic/gin.(*Context).Next
2024-06-15T10:48:39.9789211Z /home/runner/go/pkg/mod/github.com/gin-gonic/gin@v1.10.0/context.go:185
2024-06-15T10:48:39.9789842Z github.com/gin-gonic/gin.(*Engine).handleHTTPRequest
2024-06-15T10:48:39.9790467Z /home/runner/go/pkg/mod/github.com/gin-gonic/gin@v1.10.0/gin.go:633
2024-06-15T10:48:39.9791001Z github.com/gin-gonic/gin.(*Engine).ServeHTTP
2024-06-15T10:48:39.9791565Z /home/runner/go/pkg/mod/github.com/gin-gonic/gin@v1.10.0/gin.go:589
2024-06-15T10:48:39.9792223Z github.com/dana-team/platform-backend/src/routes/v1.Test_Login.func1
2024-06-15T10:48:39.9792974Z /home/runner/work/platform-backend/platform-backend/src/routes/v1/login_test.go:109
2024-06-15T10:48:39.9793497Z testing.tRunner
2024-06-15T10:48:39.9793873Z /opt/hostedtoolcache/go/1.22.2/x64/src/testing/testing.go:1689
2024-06-15T10:48:39.9794345Z --- PASS: Test_Login (0.00s)
2024-06-15T10:48:39.9794872Z --- PASS: Test_Login/ShouldFailWithInternalServerError (0.00s)
2024-06-15T10:48:39.9795492Z --- PASS: Test_Login/ShouldSuccessObtainingToken (0.00s)
2024-06-15T10:48:39.9796091Z --- PASS: Test_Login/ShouldFailWithInvalidPayload (0.00s)
2024-06-15T10:48:39.9796714Z --- PASS: Test_Login/ShouldFailWithInvalidCredentials (0.00s)
2024-06-15T10:48:39.9797154Z === RUN TestListNamespaces
2024-06-15T10:48:39.9798119Z [GIN] 2024/06/15 - 10:48:39 | 200 | 104.897µs | | GET "/v1/namespaces/"
2024-06-15T10:48:39.9798709Z --- PASS: TestListNamespaces (0.00s)
2024-06-15T10:48:39.9799050Z === RUN TestGetNamespace
2024-06-15T10:48:39.9799651Z [GIN] 2024/06/15 - 10:48:39 | 200 | 22.903µs | | GET "/v1/namespaces/test-namespace"
2024-06-15T10:48:39.9800225Z --- PASS: TestGetNamespace (0.00s)
2024-06-15T10:48:39.9800551Z === RUN TestCreateNamespace
2024-06-15T10:48:39.9801108Z [GIN] 2024/06/15 - 10:48:39 | 200 | 21.851µs | | POST "/v1/namespaces/"
2024-06-15T10:48:39.9801687Z --- PASS: TestCreateNamespace (0.00s)
2024-06-15T10:48:39.9802026Z === RUN TestDeleteNamespace
2024-06-15T10:48:39.9802629Z [GIN] 2024/06/15 - 10:48:39 | 200 | 21.901µs | | DELETE "/v1/namespaces/test-namespace"
2024-06-15T10:48:39.9803250Z --- PASS: TestDeleteNamespace (0.00s)
2024-06-15T10:48:39.9803582Z === RUN TestCreateSecret
2024-06-15T10:48:39.9804184Z [GIN] 2024/06/15 - 10:48:39 | 200 | 27.502µs | | POST "/v1/namespaces/default/secrets/"
2024-06-15T10:48:39.9804762Z --- PASS: TestCreateSecret (0.00s)
2024-06-15T10:48:39.9805070Z === RUN TestGetSecrets
2024-06-15T10:48:39.9805646Z [GIN] 2024/06/15 - 10:48:39 | 200 | 54.492µs | | GET "/v1/namespaces/default/secrets/"
2024-06-15T10:48:39.9806208Z --- PASS: TestGetSecrets (0.00s)
2024-06-15T10:48:39.9806517Z === RUN TestGetSpecificSecret
2024-06-15T10:48:39.9807185Z [GIN] 2024/06/15 - 10:48:39 | 200 | 24.315µs | | GET "/v1/namespaces/default/secrets/test-secret"
2024-06-15T10:48:39.9807808Z --- PASS: TestGetSpecificSecret (0.00s)
2024-06-15T10:48:39.9808143Z === RUN TestPatchSecret
2024-06-15T10:48:39.9808790Z [GIN] 2024/06/15 - 10:48:39 | 200 | 274.745µs | | PATCH "/v1/namespaces/default/secrets/test-secret"
2024-06-15T10:48:39.9809396Z --- PASS: TestPatchSecret (0.00s)
2024-06-15T10:48:39.9809706Z === RUN TestDeleteSecret
2024-06-15T10:48:39.9810329Z [GIN] 2024/06/15 - 10:48:39 | 200 | 16.491µs | | DELETE "/v1/namespaces/default/secrets/test-secret"
2024-06-15T10:48:39.9810934Z --- PASS: TestDeleteSecret (0.00s)
2024-06-15T10:48:39.9811228Z PASS
2024-06-15T10:48:39.9811443Z coverage: 69.2% of statements
2024-06-15T10:48:39.9812277Z ok github.com/dana-team/platform-backend/src/routes/v1 0.019s coverage: 69.2% of statements
2024-06-15T10:48:39.9812835Z FAIL
2024-06-15T10:48:40.0268467Z make: *** [Makefile:22: test] Error 1
2024-06-15T10:48:40.0286468Z ##[error]Process completed with exit code 2.
```

/kind tests

--- a/src/auth/openshift_test.go
+++ b/src/auth/openshift_test.go
@@ -12,31 +12,6 @@ import (
 )
 
 func TestObtainOpenshiftToken(t *testing.T) {
-	_ = os.Setenv(envInsecureSkipVerify, "true")
-	_ = os.Setenv(envKubeClientID, "clientid")
-
-	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		if r.URL.Path == "/auth" {
-			w.Header().Set("Location", "https://example.com/callback?code=testcode")
-			w.WriteHeader(http.StatusFound)
-			return
-		}
-
-		if r.URL.Path == "/token" {
-			if r.FormValue("code") == "testcode" {
-				w.Header().Set("Content-Type", "application/json")
-				_, _ = w.Write([]byte(`{"access_token": "test_access_token", "token_type": "bearer"}`))
-			} else {
-				w.WriteHeader(http.StatusBadRequest)
-			}
-			return
-		}
-	}))
-	defer ts.Close()
-
-	_ = os.Setenv(envKubeAuthURL, ts.URL+"/auth")
-	_ = os.Setenv(envKubeTokenURL, ts.URL+"/token")
-
 	type args struct {
 		username string
 		password string
@@ -84,8 +59,32 @@ func TestObtainOpenshiftToken(t *testing.T) {
 		},
 	}
 
+	_ = os.Setenv(envInsecureSkipVerify, "true")
+	_ = os.Setenv(envKubeClientID, "clientid")
 	for name, tc := range cases {
 		t.Run(name, func(t *testing.T) {
+			ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				if r.URL.Path == "/auth" {
+					w.Header().Set(httpLocationHeader, "https://example.com/callback?code=testcode")
+					w.WriteHeader(http.StatusFound)
+					return
+				}
+
+				if r.URL.Path == "/token" {
+					if r.FormValue(keyCodeParam) == "testcode" {
+						w.Header().Set("Content-Type", "application/json")
+						_, _ = w.Write([]byte(`{"access_token": "test_access_token", "token_type": "bearer"}`))
+					} else {
+						w.WriteHeader(http.StatusBadRequest)
+					}
+					return
+				}
+			}))
+			defer ts.Close()
+
+			_ = os.Setenv(envKubeAuthURL, ts.URL+"/auth")
+			_ = os.Setenv(envKubeTokenURL, ts.URL+"/token")
+
 			if tc.want.mockServerStatus != 0 {
 				ts.Config.Handler = http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 					w.WriteHeader(tc.want.mockServerStatus)

